homeassistant/components/waze_travel_time/sensor.py

```diff
@@ -197,41 +203,37 @@ async def async_added_to_hass(self) -> None:
             await self.first_update()
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
+    def native_value(self) -> float | None:
         """Return the state of the sensor."""
         if self._waze_data.duration is not None:
             return round(self._waze_data.duration)
 
         return None
 
     @property
-    def native_unit_of_measurement(self):
-        """Return the unit of measurement."""
-        return TIME_MINUTES
-
-    @property
-    def icon(self):
+    def icon(self) -> str:
         """Icon to use in the frontend, if any."""
-        return ICON
+        if self._waze_data.vehicle_type == "taxi":
+            return "mdi:taxi"
+        if self._waze_data.vehicle_type == "motorcycle":
+            return "mdi:motorbike"
+
+        return "mdi:car"
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict | None:
         """Return the state attributes of the last update."""
         if self._waze_data.duration is None:
             return None
 
-        res = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        res[ATTR_DURATION] = self._waze_data.duration
-        res[ATTR_DISTANCE] = self._waze_data.distance
-        res[ATTR_ROUTE] = self._waze_data.route
-        res[ATTR_ORIGIN] = self._waze_data.origin
-        res[ATTR_DESTINATION] = self._waze_data.destination
-        return res
+        return {
+            ATTR_ATTRIBUTION: "Powered by Waze",
+            "duration": self._waze_data.duration,
+            "distance": self._waze_data.distance,
+            "route": self._waze_data.route,
+            "origin": self._waze_data.origin,
+            "destination": self._waze_data.destination,
```

Comment:
I am not sure if we should drop the usage of consts here

Reply:
I think the only reason to have declared constants is when you use them in multiple places.
Pointing to a one time used constant can sometimes make it less readable.
I can even go further and say that it can be architectural problem as people can use different const name for the same word, which can be confusing.

Reply:
ok, your point makes sense